oras/provider.py

```diff
@@ -236,8 +237,31 @@ def get_tags(
         :param N: number of tags
         :type N: int
         """
-        tags_url = f"{self.prefix}://{container.tags_url(N)}"  # type: ignore
-        return self.do_request(tags_url, "GET", headers=self.headers)
+        tags_url = f"{self.prefix}://{container.tags_url(N=N)}"  # type: ignore
```

Comment:
We could probably verify this fairly quickly by arbitrarily pushing a ton of tags to the test registry in the actions and then retrieving them in this manner.

Is this pagination technique common, and if so, could it be generalized to a common function use by get tags? :thinking:

Reply:
I tried adding a test but the local registry returns all tags no matter what :) I found the docker specs though: https://docs.docker.com/registry/spec/api/#listing-image-tags where it's mentioning the pagination and some other discussion on github: 
https://github.com/orgs/community/discussions/25658 So that seems to be reasonably standard. 

I tried to find a configuration in the docker registry settings so that we could reproduce the pagination locally, but didn't find anything. 

We could test against the channel-mirrors package if that's something you'd want.

Reply:
I added a small test that uses ghcr and that arrow-cpp package with > 1000 tags.